commons/src/main/java/fr/free/nrw/commons/upload/VolleyRequestQueue.java

```diff
@@ -0,0 +1,42 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.toolbox.ImageLoader;
+import com.android.volley.toolbox.Volley;
+
+
+public class VolleyRequestQueue {
+    private static VolleyRequestQueue mInstance;
+    private RequestQueue mRequestQueue;
+    private ImageLoader mImageLoader;
+    private static Context mContext;
+
+    private VolleyRequestQueue(Context context) {
+        mContext = context;
+        mRequestQueue = getRequestQueue();
+    }
+
+    public static synchronized VolleyRequestQueue getInstance(Context context) {
+        if (mInstance == null) {
+            mInstance = new VolleyRequestQueue(context);
+        }
+        return mInstance;
+    }
+
+    public RequestQueue getRequestQueue() {
+        if (mRequestQueue == null) {
+            // getApplicationContext() is key, it keeps you from leaking the
+            // Activity or BroadcastReceiver if someone passes one in.
+            mRequestQueue = Volley.newRequestQueue(mContext.getApplicationContext());
```

Comment:
I think this would be more appropriate in the constructor. At any rate, I don't think we want to expose it.